kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -221,4 +223,38 @@ public static KernelException invalidConfigurationValueException(
     private static String formatTimestamp(long millisSinceEpochUTC) {
         return new Timestamp(millisSinceEpochUTC).toInstant().toString();
     }
+
+    // We use the `Supplier` interface to avoid silently wrapping any checked exceptions
```

Comment:
I explicitly chose to not wrap checked exceptions because (1) I think it's antithetical to the whole point of checked exceptions and (2) even just in this PR doing so had multiple downstream unexpected consequences through chained-method calls since we use exceptions for control flow in places.